src/app/lab/components/city-dropdown.component.ts

```diff
@@ -12,47 +19,151 @@ import { apiHost } from '../../constants';
               </div>
-            </div>`
+            </div>`,
+  providers: [{
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => CityDropdownComponent),
+      multi: true,
+  }, {
+    provide: NG_VALIDATORS,
+    useExisting: forwardRef(() => CityDropdownComponent),
+    multi: true,
+  }]
 })
-export class CityDropdownComponent {
+export class CityDropdownComponent implements OnInit, ControlValueAccessor, Validator {
+
+    @ViewChild('input') input: ElementRef;
 
-    public apiCities: string = apiHost + '/api/city/?search=:keyword';
+    public errors: ValidationErrors = null;
 
-    @Input() projectData: ProjectData;
-    @Input() showIcon = true;
+    private autocomplete: google.maps.places.Autocomplete;
+    private onChange = (_: any) => { };
 
-    constructor() {}
+    constructor(private el: ElementRef, private mapsApiLoader: MapsAPILoader,
+                private zone: NgZone, private mapCellService: MapCellService) {}
 
-    // custom formatter to display list of options as City, State
-    public cityListFormatter(data: any): string {
-        let html = '';
-        html += data.properties.name ?
-            `<span>${data.properties.name}, ${data.properties.admin}</span>` : data;
-        return html;
+    ngOnInit() {
+      try {
+           this.setupAutocomplete();
+        } catch (error) {
+            this.mapsApiLoader.load().then(() => this.setupAutocomplete());
+        }
     }
 
-    // custom formatter to display string for selected city as City, State
-    public cityValueFormatter(data: any): string {
-        let displayValue = '';
-        if (data && data.properties) {
-            displayValue += data.properties.name + ', ' + data.properties.admin;
+    writeValue(city: any) {
+        if (city && city.properties) {
+            this.input.nativeElement.value = `${city.properties.name}, ${city.properties.admin}`;
         }
-        return displayValue;
+    }
+
+    registerOnChange(fn: any) {
+        this.onChange = fn;
+    }
+
+    // Required by interface, not used
+    registerOnTouched(fn: any) {}
+
+    validate(c: FormControl): ValidationErrors {
+        return this.errors;
+    }
+
+    private setupAutocomplete() {
+        const options = {
+            types: ['(cities)']
+        };
+        this.autocomplete = new google.maps.places.Autocomplete(this.input.nativeElement, options);
+        this.autocomplete.addListener('place_changed', () => this.onPlaceChanged());
+
+        const forceAutocompleteOff = () => {
+            // Chrome ignores 'autocomplete="off"' but will turn off autocomplete for
+            // invalid options. Google Places sets 'autocomplete="off"' regardless of
+            // what was set on the <input> before, so we need to override that in JS
+            this.input.nativeElement.autocomplete = 'forced-false';
+            this.input.nativeElement.removeEventListener('focus', forceAutocompleteOff);
```

Comment:
It looks like this work-around is no longer working... I'm going to open an issue to address this - this will also be an issue on Temperate, which this was copied from.